core/storage/blog_post/gae_models.py

```diff
@@ -0,0 +1,454 @@
+                exists already.
+        """
+        if cls.get_by_id(blog_post_id):
+            raise Exception(
+                'Blog Post ID conflict on creating new blog post summary model.'
+                )
+
+        entity = cls(
+            id=blog_post_id,
+            author_id=author_id,
+            summary='',
+            title='',
+            published_on=None,
+            url_fragment='',
+            tags=[],
+            thumbnail_filename=''
+        )
+        entity.update_timestamps()
+        entity.put()
+
+        return entity
+
+    @classmethod
+    def get_blog_post_summary_models(cls, blog_post_ids):
+        """Returns a list of BlogPostSummaryModels for blog posts created by
+        the user.
+
+        Args:
+            blog_post_ids: list(str). Contains blog post ids for which the blog
+                post summary models are to be fetched.
+
+        Returns:
+            list(BlogPostSummaryModel). The list of BlogPostSummaryModels
+            fetched using blog_post_ids.
+        """
+        return cls.get_multi(blog_post_ids)
+
+
+class BlogPostRightsModel(base_models.BaseModel):
+    """Storage model for rights related to a blog post.
+
+    The id of each instance is the blog_post_id.
+    """
+
+    # The user_id of the blog editors of this blog post.
+    editor_ids = datastore_services.StringProperty(indexed=True, repeated=True)
+
+    # Whether this blog post is published or not.
+    # False if blog post is a draft, True if published.
+    blog_post_is_published = datastore_services.BooleanProperty(
+        indexed=True, required=True, default=False)
+
+    @staticmethod
+    def get_deletion_policy():
+        """Model contains data to pseudonymize or delete corresponding
+        to a user: editor_ids field.
+        """
+        return base_models.DELETION_POLICY.LOCALLY_PSEUDONYMIZE
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check whether BlogPostRightsModel references user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return cls.query(
+            # This checks if any of the editor IDs is equal to the user_id.
+            cls.editor_ids == user_id
+        ).get(keys_only=True) is not None
+
+    @classmethod
+    def get_by_user(cls, user_id):
+        """Retrieves the rights object for all blog posts assigned to given user
+
+        Args:
+            user_id: str. ID of user.
+
+        Returns:
+            list(BlogPostRightsModel). The list of BlogPostRightsModel objects
+            in which the given user is a editor.
+        """
+
+        return cls.query(cls.editor_ids == user_id).fetch()
+
+    @staticmethod
+    def get_model_association_to_user():
+        """Model is exported as one instance shared across users since multiple
+        users can edit the blog.
```

Comment:
I don't understand this description, multiple users can edit blog or blog post?

Reply:
Sorry, it should be blog post. Multiple users can edit the blog post. Fixed